.github/workflows/on_propen_push_do_test.yml

```diff
@@ -6,8 +6,24 @@ on:
       - main
       - 15.[0-9]+.x
       - 14.0.x
+
+concurrency:
+  # Only cancel jobs for PR updates
+  group: ci-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
+
+  build:
```

Comment:
I will need to move this in a separate workflow if we want to build the images before the tests. but this is for a future step

Reply:
Sure. I kept the build part with the tests for simplicity, but we can easily split into multiple jobs.